### PR TITLE
Editor link icon open modal

### PR DESCRIPTION
**Pull request by MuhammadKaleem171**
Editor link icon should open modal

Closes: #8553 

## Description of Changes
- Created the new modal to add link and text

## "How We Fixed It"
Created the new modal and added two input for text and link .

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx
Comment (Israellund) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx:287]: Please delete all of these comments in handleAddLink. We usually only leave comments for functions or code that isn't quickly apparent as to what they do, or if it seems unintuitive but is necessary

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx
Comment (Israellund) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx:45]: Please make this placeholder text sentence case, that is the design convention we are using across the platform.

**Review by Israellund — APPROVED** [review]
@MuhammadKaleem171 Looks good and works very well! I just asked for some small changes. I'm approving this now but please try to fix the failing devnet tests before trying to merge. If you're unable to see the details of why it's failing just let me know. Also for any future PRs, please assign me and @mzparacha @masvelio and @burtonator as reviewers, as we require at least 2 approvals from other engineers for the PR to be merged.

**Comment by MuhammadKaleem171** [comment]
"@Israellund I'm trying to troubleshoot a CI/CD error and would appreciate your help in understanding the issue. Could you assist me in identifying and resolving the problem?"

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx
Comment (Israellund) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx:45]: @MuhammadKaleem171 Should be "Link text", only the first letter of the first word

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx
Comment (MuhammadKaleem171) [reply packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx]: i have made the placeholder text sentence case

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx
Comment (MuhammadKaleem171) [reply packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx:45]: i have changed the Link Text to link text

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx:21]: We have a rule that name of the file, component, props and scss are all the same.

So LinkModal.tsx, LinkModal.scss const LinkModal  = () => {}, LinkModalProps etc

Please follow that rule in upcoming PRs

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx:407]: take a look on other modal implementations.
1. We use CWModal component and pass `open` prop 
2. as a `content` prop we pass a component 

image

**Review by masvelio — COMMENTED** [review]
it works well but please adjust the code according to my comments. It will be crucial to be consistent with existing codebase, just to prevent mess. Thanks for contribution!

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx
Comment (MuhammadKaleem171) [reply packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx]: I follow the above example and use the CWModal and pass the content as component

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx
Comment (MuhammadKaleem171) [reply packages/commonwealth/client/scripts/views/components/react_quill_editor/QuillLinkModal.tsx]: i change the AddLinkModal to LinkModal and file name to LinkModal.tsx

File: packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx
Comment (MuhammadKaleem171) [reply packages/commonwealth/client/scripts/views/components/react_quill_editor/react_quill_editor.tsx]: While testing the change, I discovered a bug: when the modal is closed, the cursor moves to the beginning of the input field. I’ve also made improvements to address this issue.

**Review by masvelio — APPROVED** [review]
Thanks for changes, please follow my advices in the future. Good start!

**Comment by MuhammadKaleem171** [comment]
@masvelio @Israellund should i close the conversation ?

**Comment by masvelio** [comment]
> @masvelio @Israellund should i close the conversation ?

@MuhammadKaleem171 thanks for giving acknowledge to the specific comments that you fixed the issue. If it not requires further discussion, you can resolve it.

**Review by Israellund — APPROVED** [review]

**Pull request closed without merging**
